### PR TITLE
Use each IP to probe public and private services

### PR DESCRIPTION
**Pull request by nak3**
Currently prober code probes public (`hello.example.com`) and private (`hello.default.svc.cluster.local`) services by ingress's private endpoint IP only. 

i.e.
- `-H "hello.example.com"` `<private ingress's IP>`
- `-H "hello.default.svc.cluster.local"` `<private ingress's IP>`

This probe code works with Istio and Kourier as they map two services (local and external) to one deployment.
However, it does not work with Contour as it maps one service to one deployment.

So, this patch fixes the prober code to use each public and private IPs for external and local service.

i.e.
- `-H "hello.example.com"` `<public inigress's IP>`
- `-H "hello.default.svc.cluster.local"` `<private inigress's IP>`

Part of https://github.com/knative-sandbox/net-gateway-api/pull/179

### CODE REVIEW

File: third_party/contour-head/gateway/gateway-internal.yaml [hunk 1]
```diff
@@ -65,10 +70,6 @@ spec:
         from: "All"
   - protocol: HTTPS
     port: 443
-    tls:
-      mode: Terminate
-      routeOverride:
-        certificate: Allow
```
Comment (markusthoemmes) [inline third_party/contour-head/gateway/gateway-internal.yaml:71]: Are these changes related?

Comment (nak3) [reply third_party/contour-head/gateway/gateway-internal.yaml]: It is not related. I reverted it to be clear.

File: pkg/reconciler/ingress/lister.go [hunk 2]
```diff
@@ -52,12 +52,26 @@ type gatewayPodTargetLister struct {
 }
 
 func (l *gatewayPodTargetLister) ListProbeTargets(ctx context.Context, ing *v1alpha1.Ingress) ([]status.ProbeTarget, error) {
+	var err error
+	visibilityToEndpointIPs := map[v1alpha1.IngressVisibility]sets.String{}
```
[the remark this answers is not in the document]
Comment (nak3) [reply pkg/reconciler/ingress/lister.go]: Sounds good. Updated.

**Comment by nak3** [comment]
Thank you! Updated.

File: pkg/reconciler/ingress/lister.go [hunk 3]
```diff
@@ -52,12 +52,26 @@ type gatewayPodTargetLister struct {
...
 func (l *gatewayPodTargetLister) ListProbeTargets(ctx context.Context, ing *v1alpha1.Ingress) ([]status.ProbeTarget, error) {
+	var err error
+	var privateIPs, publicIPs sets.String
+
+	if privateIPs, err = l.endpointIPs(ctx, v1alpha1.IngressVisibilityClusterLocal); err != nil {
+		return nil, err
+	}
+	if publicIPs, err = l.endpointIPs(ctx, v1alpha1.IngressVisibilityExternalIP); err != nil {
+		return nil, err
+	}
```
Comment (markusthoemmes) [inline pkg/reconciler/ingress/lister.go:63]: nit:

```suggestion
	privateIPs, err := l.endpointIPs(ctx, v1alpha1.IngressVisibilityClusterLocal)
	if err != nil {
		return nil, err
	}
	publicIPs, err := l.endpointIPs(ctx, v1alpha1.IngressVisibilityExternalIP)
	if err != nil {
		return nil, err
	}
```

**Review by markusthoemmes — APPROVED** [review]
/lgtm
/approve

/hold

If you want to fix the nit. Feel free to just unhold.

**Review by markusthoemmes — APPROVED** [review]
/lgtm
/approve
